rabbitmq/rabbitmq-stream-dotnet-client
# New event to handle Metadata update

With this PR https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/328 the client can handle multi-producers and consumers per connection.

This PR removes [`MetadataHandler`](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/332/files#diff-71215640c5968601ad0a124946241692beb3c931cc450bb43bb7061e8f84c47eL56)  and introduces [`OnMetadataUpdate`](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/332/files#diff-71215640c5968601ad0a124946241692beb3c931cc450bb43bb7061e8f84c47eR58
) event.

The event can handle multiple Metadata updates coming from the server. Metadata update is raised when a stream is deleted, or a replica is removed. 

The server automatically removes the producers and consumers linked to the connection, here we need to remove these entities from the internal pool to be consistent. 

How to test
---
- You need a cluster 
- Use `RawProducer` and `RawConsumer` bind to one or more streams. Delete the streams and the client has to remove all the connections 
- Use `Producer` and `Consumer` in a cluster remove a replica the `Producer` and `Consumer` should reconnect and continue to work 


Refactor
---

Refactor `RawConsumer` and `RawProducer` in [this commit](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/332/commits/c5847829ce6d8491fd5f239e016a54376b03818c). Remove duplication code. Move the common code to the `AbstractEntity` Class
